File: java/src/patterns/creational/singleton/EagerSingleton.java
package patterns.creational.singleton;

class EagerSingleton {
    // single static instance instantiated eagerly
    private static final EagerSingleton instance = new EagerSingleton();

    // private constructor prevents object instantiation
    private EagerSingleton() {
        // init logic here
    }

    public static EagerSingleton getInstance() {
        return instance;
    }
}

File: python/src/patterns/creational/singleton/eager_singleton.py
class EagerSingleton:
    _instance = None

    def __init__(self) -> None:
        # prevent multiple instantiation by raising an exception
        if EagerSingleton._instance is not None:
            raise Exception("Cannot instantiate directly.")
        # init logic here

    @staticmethod
    def get_instance() -> "EagerSingleton":
        return EagerSingleton._instance

# create instance immediately on module import
EagerSingleton._instance = EagerSingleton()
